test/replication-luatest/gh_7512_raft_leader_hang_test.lua

```diff
@@ -0,0 +1,67 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local fio = require('fio')
+
+local g = t.group('gh-7512')
+
+-- gh-7512: Raft followers should notice leader hang and arrange new elections.
+g.before_all(function(cg)
+    cg.cluster = cluster:new{}
+    local box_cfg = {
+        election_mode = 'candidate',
+        replication_timeout = 0.1,
+        replication_synchro_quorum = 1,
+        replication = {
+            server.build_instance_uri('node1'),
+            server.build_instance_uri('node2'),
+        },
+    }
+    cg.node1 = cg.cluster:build_and_add_server{
+        alias = 'node1',
+        box_cfg = box_cfg,
+    }
+    box_cfg.election_mode='voter'
+    cg.node2  = cg.cluster:build_and_add_server{
+        alias = 'node2',
+        box_cfg = box_cfg,
+    }
+    cg.cluster:start()
+
+    cg.unique_filename = server.build_instance_uri('unique_filename')
+    fio.unlink(cg.unique_filename)
+end)
+
+g.after_all(function(cg)
+    cg.cluster:drop()
+    fio.unlink(cg.unique_filename)
+end)
+
+local function block(filename)
+    local fiber = require('fiber')
+    fiber.new(function(filename)
+        -- Can't use fio, because it yields.
+        local cmd = 'sleep 0.1 && ls ' .. filename .. ' > /dev/null 2>&1'
+        while os.execute(cmd) ~= 0 do end
```

Comment:
You might want to turn this into `luatest.helpers.retrying()` instead of the `while` loop. Otherwise in case of any problem we will get the process which refuses to terminate until `kill -9` in a console. Test-run is really bad at killing such stubborn busy-looped processes.

Reply:
`helpers.retrying` yields in between retries, so it would unblock the tx thread once in a while and spoil the test.
AFAICS I could only do something like `while os.execute(cmd) ~= 0` **and** `clock.time() - start < timeout` `do end`.
But this wouldn't help much, because if the timeout is too small, the test might flaky fail, and if it's too big, the test might already be finished by the time timeout passes, so test-run will fail at stopping the node anyway.

Reply:
Thanks for the review and for all the suggestions!